15-workflow-sets.Rmd

```diff
@@ -0,0 +1,441 @@
+
+```{r workflow-sets-startup, include = FALSE}
+knitr::opts_chunk$set(fig.path = "figures/")
+library(tidymodels)
+library(workflowsets)
+library(baguette)
+library(rules)
+library(finetune)
+
+caching <- FALSE
+
+cores <- parallel::detectCores()
+if (!grepl("mingw32", R.Version()$platform)) {
+ library(doMC)
+ registerDoMC(cores = cores)
+} else {
+  library(doParallel)
+  cl <- makePSOCKcluster(cores)
+  registerDoParallel(cl)
+}
+```
+
+# Screening many models  {#workflow-sets}
+
+Workflow sets have been previously discussed in Chapters \@ref(workflows) and \@ref(compare). In this chapter, we discuss them in more detail and describe a use case where they can be helpful. 
+
+For projects where the data are novel, there may be the need to screen many combinations of models and preprocessors. There might not be any _a priori_ knowledge about which methods work best. 
+
+:::rmdnote
+A good strategy would be to try a variety of approaches, determine what deems to work best, then spend additional time tweaking/optimizing a small set of models.   
+:::
+
+Workflow sets provide a nice user interface to create and manage this process. We'll also demonstrate how to evaluate these models efficiently using the racing methods discussed in Section \@ref(racing-example).
+
+## Modeling concrete strength
+
+Let's use the concrete data from _Applied Predictive Modeling_ [@apm] as an example. In Chapter 10 of that book, models predicted the compressive strength of mixtures using the ingredients as predictors. A wide variety of models were evaluated with different predictor sets and preprocessing needs.  How can workflow sets make the large scale testing of models easier? 
+
+First, let's define the data splitting and resampling schemes. The data are in the `RData` directory of the GitHub repository: 
+
+```{r workflow-sets-data}
+library(tidymodels)
+load("RData/concrete_mixtures.RData")
+glimpse(concrete_mixtures)
```

Comment:
This isn't super accessible for folks or easy to follow along with, but it's very close to what is in [modeldata](https://modeldata.tidymodels.org/reference/concrete.html), right? Can we use that? Is there a reason not to? There are no factor levels so I'd prefer to see a CSV to a URL if we can't use that.

Reply:
I'm going to see if this version of the data first into `modeldata`. We could use what's already there but this is the version that I use in _APM_ and it would be good to be consistent.

Reply:
OK, let's see if it will fit in modeldata and if not use a CSV. Or at the least, an `.rds` file. Loading a whole `.Rdata` here feels weird to me.